packages/components/src/range-control/styles/range-control-styles.js

```diff
@@ -0,0 +1,310 @@
+	align-items: center;
+	margin-left: -10px;
+	margin-top: 5px;
+	width: 20px;
+	box-sizing: border-box;
+	display: flex;
+	height: 20px;
+	justify-content: center;
+	outline: 0;
+	position: absolute;
+	pointer-events: none;
+	top: 0;
+	user-select: none;
+
+	[dir='rtl'] & {
+		margin-left: 0;
+		margin-right: -10px;
+	}
+`;
+
+const handleReducedMotion = () => {
+	return css`
+		@media ( prefers-reduced-motion: reduce ) {
+			transition: none !important;
+		}
+	`;
+};
+
+const thumbFocus = ( { isFocused } ) => {
+	return css( {
+		borderColor: isFocused
+			? color( 'blue.medium.focus' )
+			: color( 'darkGray.200' ),
+		boxShadow: isFocused
+			? `
+				0 0 0 1px ${ color( 'blue.medium.focus' ) }
+			`
+			: `
+				0 0 0 rgba(0, 0, 0, 0)
+			`,
+	} );
+};
+
+export const Thumb = styled.span`
+	align-items: center;
+	background-color: white;
+	border-radius: 50%;
+	border: 1px solid ${color( 'darkGray.200' )};
+	box-sizing: border-box;
+	height: 100%;
+	outline: 0;
+	pointer-events: none;
+	position: absolute;
+	user-select: none;
+	width: 100%;
+
+	${thumbFocus};
+`;
+
+export const InputRange = styled.input`
+	box-sizing: border-box;
+	cursor: pointer;
+	display: block;
+	height: 100%;
+	left: 0;
+	margin: 0;
+	opacity: 0;
+	outline: none;
+	position: absolute;
+	right: 0;
+	top: 0;
+	width: 100%;
+`;
+
+const tooltipShow = ( { show } ) => {
+	return css( {
+		opacity: show ? 1 : 0,
+	} );
+};
+
+const tooltipPosition = ( { position } ) => {
+	const isTop = position === 'top';
+
+	if ( isTop ) {
+		return css`
+			margin-top: -4px;
+			top: -100%;
+
+			&::after {
+				border-bottom: none;
+				border-top-style: solid;
+				bottom: -6px;
+			}
+		`;
+	}
+
+	return css`
+		margin-bottom: -4px;
+		bottom: -100%;
+
+		&::after {
+			border-bottom-style: solid;
+			border-top: none;
+			top: -6px;
+		}
+	`;
+};
+
+export const Tooltip = styled.span`
+	background: ${color( 'darkGray.800' )};
+	border-radius: 3px;
+	box-sizing: border-box;
+	color: white;
+	display: inline-block;
+	font-size: 11px;
+	min-width: 32px;
+	opacity: 0;
+	padding: 8px;
+	position: absolute;
+	text-align: center;
+	transform: translateX( -50% );
+	transition: opacity 120ms ease;
+	user-select: none;
+
+	&::after {
+		border: 6px solid ${color( 'darkGray.800' )};
+		border-left-color: transparent;
+		border-right-color: transparent;
+		bottom: -6px;
+		box-sizing: border-box;
+		content: '';
+		height: 0;
+		left: 50%;
+		line-height: 0;
+		margin-left: -6px;
+		position: absolute;
+		width: 0;
+	}
+
+	${tooltipShow};
+	${tooltipPosition};
+	${handleReducedMotion};
+
+	[dir='rtl'] & {
+		transform: translateX( 50% );
+	}
+`;
+
+export const InputNumber = styled.input`
+	box-sizing: border-box;
+	display: inline-block;
+	margin-left: 16px;
+	margin-top: 0;
+	min-width: 54px;
+	max-width: 120px;
+
+	input[type='number']& {
+		${rangeHeight};
+	}
+`;
+
+export const ActionRightWrapper = styled.span`
+	box-sizing: border-box;
+	display: block;
+	margin-left: 8px;
```

Comment:
Each time I see margin-left (or right or padding), I wonder whether we need RTL equivalents. Maybe we can solve these with mixins for now, otherwise, we'll quickly forget about these.

Reply:
What about this one, does it need to be wrapped in rtl?